gnocchi/rest/api.py

```diff
@@ -148,18 +153,23 @@ def deserialize(expected_content_types=None):
     return params
 
 
-def validate(schema, data, required=True):
+def validate(schema, data, required=True, detailed_exc=False):
     try:
         return voluptuous.Schema(schema, required=required)(data)
     except voluptuous.Error as e:
-        abort(400, "Invalid input: %s" % e)
+        if detailed_exc:
+            abort(400, e)
```

Comment:
But this is gonna return a string right, because `voluptuous.Error` has no jsonify()? This does not seem like the final goal we want to achieve.

Reply:
abort() method build the json for any voluptuous error now.

Reply:
I see, I will move the json build here, it make more sense